CMakeLists.txt

```diff
@@ -18,6 +18,13 @@ INCLUDE(cmake/base.cmake)
 INCLUDE(cmake/boost.cmake)
 INCLUDE(cmake/apple.cmake)
 INCLUDE(cmake/stubs.cmake)
+IF(APPLE) # Use the handmade approach
+  SET(CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake/find-external/OpenMP ${CMAKE_MODULE_PATH})
+ELSEIF(UNIX)
+  IF(${CMAKE_VERSION} VERSION_GREATER "3.20.0" OR ${CMAKE_VERSION} VERSION_EQUAL "3.20.0")
+    set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake/find-external/OpenMP ${CMAKE_MODULE_PATH})
+  ENDIF()
```

Comment:
Why are we handling this here instead of the cmake submodule?

Also, it seems we aren't including OpenMP when cmake is lower than 3.20. Why are we doing so?

Is there something that I am missing?